src/axom/core/tests/core_array.hpp

```diff
@@ -1583,4 +1583,76 @@ TEST(core_array, checkDevice2D)
 #endif
 }
 
+struct HasDefault
+{
+  int member = 255;
+};
+
```

Comment:
It might be good to add another test for the basic case, e.g. `Array<int> arr(10)` that tests that the array is default-initialized to `0`s.